nav2_behavior_tree/plugins/action/remove_passed_goals_action.cpp

```diff
@@ -67,18 +67,44 @@ inline BT::NodeStatus RemovePassedGoals::tick()
     return BT::NodeStatus::FAILURE;
   }
 
+  // get the `waypoint_statuses` vector
+  std::vector<nav2_msgs::msg::WaypointStatus> waypoint_statuses;
+  auto waypoint_statuses_get_res = getInput("input_waypoint_statuses", waypoint_statuses);
+  if (!waypoint_statuses_get_res) {
+    RCLCPP_WARN(node_->get_logger(), "Missing [input_waypoint_statuses] port input!");
+  }
+
   double dist_to_goal;
-  while (goal_poses.goals.size() > 1) {
+  while (goal_poses.goals.size() > 0) {
```

Comment:
This could be changed to a `do while` rather than a `while` to check the condition at the end to keep the `size() > 1` and not need the addition `if` check on it to `break`